packages/angular/src/utils/mfe/with-module-federation.ts

```diff
@@ -0,0 +1,222 @@
+import {
+  SharedLibraryConfig,
+  sharePackages,
+  shareWorkspaceLibraries,
+} from './mfe-webpack';
+import {
+  createProjectGraphAsync,
+  readCachedProjectGraph,
+} from '@nrwl/workspace/src/core/project-graph';
+import { readWorkspaceJson } from '@nrwl/workspace';
+import { joinPathFragments, ProjectGraph } from '@nrwl/devkit';
+import ModuleFederationPlugin = require('webpack/lib/container/ModuleFederationPlugin');
+import { Workspaces } from 'nx/src/shared/workspace';
+import { appRootPath } from 'nx/src/utils/app-root';
+import { getRootTsConfigPath } from '@nrwl/workspace/src/utilities/typescript';
+import { readTsConfig } from '@nrwl/workspace';
```

Comment:
The `@nrwl/workspace` entry point is pretty heavy, importing it has perf implications. Let's change this to a deep import.